File: code/python/02_event_study.py

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from pathlib import Path

CLEAN = Path("data/clean")
DOCS = Path("docs")
DOCS.mkdir(exist_ok=True)

def main():
    panel_path = CLEAN / "country_quarter_panel.csv"
    if not panel_path.exists():
        raise SystemExit("Run 01_build_panel.py first.")
    df = pd.read_csv(panel_path)

    # TODO: compute event-time relative to first shock per country-capability
    # Placeholder: create an empty figure so the pipeline runs.
    plt.figure()
    plt.title("Event Study Placeholder")
    plt.xlabel("Event Time (quarters)")
    plt.ylabel("Outcome (e.g., PESCO in area)")
    fig_path = DOCS / "event_study_placeholder.png"
    plt.savefig(fig_path, dpi=200, bbox_inches="tight")
    print(f"Wrote {fig_path}")

if __name__ == "__main__":
    main()

File: code/R/02_event_study.R

library(tidyverse)
stopifnot(file.exists("data/clean/country_quarter_panel.csv"))
df <- readr::read_csv("data/clean/country_quarter_panel.csv")
# TODO: implement event-time computation and plot
ggplot() + ggtitle("Event Study Placeholder") +
  xlab("Event Time (quarters)") + ylab("Outcome")
ggplot2::ggsave("docs/event_study_placeholder_R.png", dpi = 200, width = 6, height = 4)
message("Wrote docs/event_study_placeholder_R.png")
